contracts/base/ChainId.sol

```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity =0.7.6;
+
+/// @title Function for getting the chainid
+/// @dev Base contract that is overridden for tests
+abstract contract ChainId {
+    /// @dev Method that exists purely to be overridden for tests
+    /// @return The chainId
+    function _chainid() internal view virtual returns (uint256) {
+        uint256 chainID;
+        assembly {
+            chainID := chainid()
```

Comment:
if you make this a one liner it will be inlined into the calling fn

also naming can just be get() and it should be a library